src/med_bench/mediation.py

```diff
@@ -433,29 +439,9 @@ def mediation_multiply_robust(y, t, m, x, interaction=False, forest=False,
         - If x, t, m, or y don't have the same length.
         - If m is not binary.
     """
-    # Format checking
-    if len(y) != len(y.ravel()):
-        raise ValueError("Multidimensional y is not supported")
-    if len(t) != len(t.ravel()):
-        raise ValueError("Multidimensional t is not supported")
-    if len(m) != len(m.ravel()):
-        raise ValueError("Multidimensional m is not supported")
-
-    n = len(y)
-    if len(x.shape) == 1:
-        x.reshape(n, 1)
-    if len(m.shape) == 1:
-        m.reshape(n, 1)
-
-    dim_m = m.shape[1]
-    if n * dim_m != sum(m.ravel() == 1) + sum(m.ravel() == 0):
-        raise ValueError("m is not binary")
+    # check input
+    y, t, m, x = _check_input(y, t, m, x, setting='binary')
```

Comment:
Please make sure that the argument order `y, t, m, x` is there in all functions. Sometimes, there is `t, m, x, y`